src/coreclr/jit/emitarm64.cpp

```diff
@@ -16367,4 +16364,83 @@ emitter::RegisterOrder emitter::IsOptimizableLdrStrWithPair(
     return optimisationOrder;
 }
 
+//-----------------------------------------------------------------------------------
+// IsOptimizableLdrToMov: Check if it is possible to optimize a second "ldr"
+//                        instruction into a cheaper "mov" instruction.
+//
+// Examples:            ldr     w1, [x20, #0x10]
+//                      ldr     w2, [x20, #0x10]    =>  mov     w1, w2
+//
+// Arguments:
+//     ins  - The instruction code
+//     reg1 - Register 1 number
+//     reg2 - Register 2 number
+//     imm  - Immediate offset, prior to scaling by operand size
+//     size - Operand size
+//     fmt  - Instruction format
+//
+// Return Value:
+//    true       - Optimization of the second instruction is possible
+//
+bool emitter::IsOptimizableLdrToMov(
+    instruction ins, regNumber reg1, regNumber reg2, ssize_t imm, emitAttr size, insFormat fmt)
+{
+    if (ins != INS_ldr)
+    {
+        // This instruction is not an "ldr" instruction.
+        return false;
+    }
+
+    if (ins != emitLastIns->idIns())
+    {
+        // Not successive "ldr" instructions.
+        return false;
+    }
+
+    regNumber prevReg1   = emitLastIns->idReg1();
+    regNumber prevReg2   = emitLastIns->idReg2();
```

Comment:
`prevReg2` here is encoded. Doesn't it need `prevReg2 = encodingZRtoSP(prevReg2)`?

(Did you see any diffs with a base register of `SP`?)

Reply:
The check is [pulled up](https://github.com/dotnet/runtime/pull/83458/files/577727e7c9938377307dd8b7816d0e8c82a0e7e3#diff-e626692803b11ee24ad5f0c575b7e591b78b661e5e56ef0bb6175554ed20e7d9R161) in the `OptimizeLdrStr()`.

Reply:
> An alternative would be to store the "actual" register in the instrDesc

Unfortunately, REG_SP is 64 and REGNUM_BITS is 6, so we don't have enough bits currently to store REG_SP in idReg1/idReg2/etc. Bumping REGNUM_BITS to 7 just to allow this might have negative effects on memory usage or other instrDesc effects.

Reply:
Thanks for the explanation @BruceForstall,  I'll add a new PR addressing the comments 👍

Reply:
Created #83886